identity/aziot-identityd/src/identity.rs

```diff
@@ -516,7 +516,11 @@ impl IdentityManager {
                             self.get_backup_provisioning_info(credential.clone())?;
 
                         if skip_if_backup_is_valid && backup_device.is_some() {
-                            backup_device.expect("backup device cannot be none")
+                            let backup_device =
```

Comment:
Can we DRY these statements into the `operation_to_iot_hub_device` method and rename the method?

Reply:
This log message is only printed when a valid backup exists; i.e. when IS is not making an operation to IoT Hub.

Reply:
I forgot to tag my comment for all the statements from lines 515-527. I meant DRY all those lines for all DPS auth types. I think that can be done?

Reply:
Got it. Did that.